cmd/livepeer/livepeer.go

```diff
@@ -183,20 +191,24 @@ func main() {
 	} else if transcoder {
 		n.NodeType = core.TranscoderNode
 		glog.Info("***Livepeer is in transcoder mode ***")
-		server.RunTranscoder(n, *orchAddr)
+		if len(orchAddresses) > 0 {
+			server.RunTranscoder(n, orchAddresses[0])
+		} else {
+			glog.Errorf("No orchestrator specified; transcoding will not happen")
+		}
 		return
 	} else {
 		n.NodeType = core.BroadcasterNode
 	}
-
 	if n.NodeType == core.BroadcasterNode {
-		if *orchAddr == "" {
-			glog.Info("No orchestrator specified; transcoding will not happen")
+		if len(orchAddresses) <= 0 && !*offchain {
+			n.OrchestratorPool = discovery.NewOnchainOrchestratorPool(n)
 		} else {
-			n.OrchestratorSelector = discovery.NewOffchainOrchestrator(n, *orchAddr)
-			if n.OrchestratorSelector == nil {
-				return
-			}
+			n.OrchestratorPool = discovery.NewOrchestratorPool(n, orchAddresses)
+		}
+		if n.OrchestratorPool == nil {
+			n.OrchestratorPool = discovery.NewOrchestratorPool(n, []string{})
```

Comment:
It should be safe to leave `OrchestratorPool` empty since we check for that specific case and return an `ErrDiscovery` if the pool is empty. That feels like a more consistent way of differentiating between the states of the program (eg, no orchestrators specified here) rather than mixing in with the possibility that some orchestrators *were* specified but none were available in a timely fashion during the `GetOrchestrators` call.

Reply:
I see why the distinction is important. We return an ErrDiscovery in `startBroadcast`, only place we actually need to call `GetOrchestrators` from the OrchestratorPool.